tests/trainer/test_trainer.py

```diff
@@ -873,6 +873,52 @@ def on_keyboard_interrupt(self, trainer, pl_module):
     assert isinstance(handle_interrupt_callback.exc_info[1], KeyboardInterrupt)
 
 
+def test_on_exception_hook(tmpdir):
+    """Test the on_exception callback hook."""
+
+    model = EvalModelTemplate()
```

Comment:
`EvalModelTemplate` is quite legacy. I believe most of the tests have switched to the BoringModel instead.

@Borda  - potentially another project for Code Health could be to remove the dependency on EvalModelTemplate in favor of either mocked lightning modules or the boring model for integration/end-to-end tests. this could also be something new contributors could help with, similar to typing. what do you think?